carbon-design-system/carbon
# feat(TimePicker): class -> functional component

REF #9712 

Refactored `TimePicker` into a functional component with RTL tests.

- Removed instances of `evt.persist();` because it is no longer needed as of React 17. However, this causes Storybook to generate console error messages when using Actions, so I removed them.

#### Testing / Reviewing

1. Pull down locally
2. Go to `packages/react`
3. Run `yarn start:v11` to run old storybook with v11 components.
4. Ensure tests for Next versions of `TimePicker` go green.
